### PR TITLE
[BUGFIX release] handle case where linked relationship data is null

### PR DESCRIPTION
**Pull request by pangratz**
This addresses #3451.

### CODE REVIEW

File: packages/ember-data/lib/serializers/json-api-serializer.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ export default JSONSerializer.extend({
 
     if (Ember.typeOf(documentHash.data) === 'object') {
       documentHash.data = this._normalizeResourceHelper(documentHash.data);
-    } else {
+    } else if (Ember.typeOf(documentHash.data) === 'array') {
```
Comment (pangratz) [inline packages/ember-data/lib/serializers/json-api-serializer.js]: @wecc would you prefer a more readable way without the `else if`, like we do in [`json-api-serializer/extractRelationship`](http://git.io/vtCxf)?

File: packages/ember-data/tests/integration/adapter/json-api-adapter-test.js [hunk 2]
```diff
@@ -875,3 +875,34 @@ test('update record - serialize hasMany', function() {
     });
   });
 });
+
+test('fetching a relationship via link works with data = null - #3451', function() {
```
Comment (pangratz) [inline packages/ember-data/tests/integration/adapter/json-api-adapter-test.js]: Not so sure about the test name but couldn't come up with a better one. Suggestions?

File: packages/ember-data/lib/serializers/json-api-serializer.js (hunk 1 above)
Comment (wecc) [inline packages/ember-data/lib/serializers/json-api-serializer.js]: This seems fine

File: packages/ember-data/tests/integration/adapter/json-api-adapter-test.js [hunk 3]
```diff
@@ -875,3 +875,37 @@ test('update record - serialize hasMany', function() {
...
+    data: {
+      type: 'post',
+      id: '1',
+      attributes: {
+        title: 'Ember.js rocks'
+      },
+      relationships: {
+        author: {
+          links: {
+            related: 'http://example.com/author/1'
```
Comment (wecc) [inline packages/ember-data/tests/integration/adapter/json-api-adapter-test.js]: Can we change this URL to `http://example.com/post/1/author`? It really doesn't make sense to provide a URL to a specific author with an ID and then return `null`.

Sorry for the OCD.

File: packages/ember-data/tests/integration/adapter/json-api-adapter-test.js (hunk 2 above)
Comment (wecc) [inline packages/ember-data/tests/integration/adapter/json-api-adapter-test.js]: Maybe something like `fetching a belongsTo relationship link that returns null`

**Comment by wecc** [comment]
`[BUGFIX beta]`?

:+1:

**Comment by pangratz** [comment]
> Edit: Hm, maybe should to into release too while we're at it.

You mean a PR into the `release` branch, right?

**Comment by wecc** [comment]
> You mean a PR into the release branch, right?

No, this is fine, against `mater`. But `[BUGFIX release]` and we can cherry-pick to both `release` and `beta`.

**Comment by pangratz** [comment]
> No, this is fine, against mater. But [BUGFIX release] and we can cherry-pick to both release and beta.

Gotcha. Done.

**Comment by wecc** [comment]
Thanks @pangratz!

**Pull request merged**